### PR TITLE
GEODE-10328: close data store stats on cache close

### PR DESCRIPTION
**Pull request by dschneider-pivotal**
### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### CODE REVIEW

File: geode-core/src/main/java/org/apache/geode/internal/cache/PartitionedRegion.java [hunk 1]
```diff
@@ -7767,6 +7768,16 @@ protected void postDestroyRegion(boolean destroyDiskRegion, RegionEventImpl even
         cache.getInternalDistributedSystem().getDistributedMember(), null, op.isClose());
   }
 
+  private void closeDataStoreStats() {
+    PartitionedRegionDataStore localDataStore = dataStore;
```
Comment (DonalEvans) [inline geode-core/src/main/java/org/apache/geode/internal/cache/PartitionedRegion.java:7772]: Is this local variable necessary? Could we just use `dataStore` directly in this method?

**Review by DonalEvans — COMMENTED** [review]
Would it be possible to add a unit or integration test covering the new behaviour, to confirm that we close the stats as expected?

File: geode-core/src/main/java/org/apache/geode/internal/cache/PartitionedRegion.java (hunk 1 above)
Comment (dschneider-pivotal) [reply geode-core/src/main/java/org/apache/geode/internal/cache/PartitionedRegion.java:7772]: I think it is needed when reading a field that is not final. I know we currently only ever set it to non-null but I just wanted this code to be future proof against any possible future changes that could modify this non-final field.

Comment (dschneider-pivotal) [reply geode-core/src/main/java/org/apache/geode/internal/cache/PartitionedRegion.java:7772]: But as a look at all the existing code in this class that simply checks if "dataStore" is null, I'll change this code to also follow that pattern.

**Review by DonalEvans — APPROVED** [review]

**Pull request merged**